docs/source/contributing/contributing.md

```diff
@@ -0,0 +1,119 @@
+# Contributing to repo2docker development
+
+## Process for making a code contribution
+
+This outlines the process for getting changes to the code of
+repo2docker merged.
+
+* If your change is relatively significant, **open an issue to discuss**
+  before spending a lot of time writing code. Getting consensus with the
+  community is a great way to save time later.
+* Make edits in your fork of the repo2docker repository
+* Submit a pull request (this is how all changes are made)
+* Edit [the changelog](https://github.com/jupyter/repo2docker/blob/master/CHANGES.rst)
+  by appending your feature / bug fix to the development version.
+* Wait for a community member to merge your changes
+* (optional) Deploy a new version of repo2docker to mybinder.org by [following these steps](http://mybinder-sre.readthedocs.io/en/latest/deployment/how.html)
+
+
+## Guidelines to getting a Pull Request merged
+
+These are not hard rules to be enforced by :police_car: but instead guidelines
+to help you make the most effictive / efficient contribution.
+
+* prefix the title of your pull request with `[MRG]` if the contribution
+  is complete and should be subjected to a detailed review;
+* create a PR as early as possible, marking it with `[WIP]` while you work on
+  it (good to avoid duplicated work, get broad review of functionality or API,
+  or seek collaborators);
+* a PR solves one problem (do not mix problems together in one PR) with the
+  minimal set of changes;
+* describe why you are proposing the changes you are proposing;
+* try to not rush changes (the definition of rush depends on how big your
+  changes are);
+* someone else has to merge your PR;
+* new code needs to come with a test;
+* apply [PEP8](https://www.python.org/dev/peps/pep-0008/) as much
+  as possible, but not too much;
+* no merging if travis is red;
+* do use merge commits instead of merge-by-squashing/-rebasing. This makes it
+  easier to find all changes since the last deployment `git log --merges --pretty=format:"%h %<(10,trunc)%an %<(15)%ar %s" <deployed-revision>..`
+
+
+## Setting up for Local Development
+
+To develop & test repo2docker locally, you need:
+
+1. Familiarity with using a command line terminal
+2. A computer running macOS / Linux
+3. Some knowledge of git
+4. At least python 3.6
+5. Your favorite text editor
+6. A recent version of [Docker Community Edition](https://www.docker.com/community-edition)
+
+### Clone the repository
+
+First, you need to get a copy of the repo2docker git repository on your local
+disk.
+
+```bash
+git clone https://github.com/jupyter/repo2docker
```

Comment:
Should we tell people to clone their own fork instead? The thing that is missing from there docs is the final step of making a PR :-/

Or we leave it for now?